.gitlab-ci.yml

```diff
@@ -0,0 +1,105 @@
+stages:
+  - lint
+  - test
+
+
+variables:                         &default-vars
+  GIT_STRATEGY:                    fetch
+  GIT_DEPTH:                       100
+  CARGO_INCREMENTAL:               0
+  CARGO_TARGET_DIR:                "/ci-cache/${CI_PROJECT_NAME}/targets/${CI_COMMIT_REF_NAME}/${CI_JOB_NAME}"
+  CI_IMAGE:                        "paritytech/ci-linux:production"
+
+.rust-info-script:                 &rust-info-script
+  - rustup show
+  - cargo --version
+  - rustup +nightly show
+  - cargo +nightly --version
+  - bash --version
+  - sccache -s
+
+.docker-env:                       &docker-env
+  image:                           "${CI_IMAGE}"
+  before_script:
+    - *rust-info-script
+    - ./scripts/ci/pre_cache.sh
+    - sccache -s
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - linux-docker
+
+.kubernetes-env:                   &kubernetes-env
+  image:                           "${CI_IMAGE}"
+  retry:
+    max: 2
+    when:
+      - runner_system_failure
+      - unknown_failure
+      - api_failure
+  interruptible:                   true
+  tags:
+    - kubernetes-parity-build
+
+.collect-artifacts:                &collect-artifacts
+  artifacts:
+    name:                          "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}"
+    when:                          on_success
+    expire_in:                     28 days
+    paths:
+      - ./artifacts/
+
+#### stage:                        lint
+
+fmt:
+  stage:                           lint
+  <<:                              *kubernetes-env
+  script:
+    - rustup component add rustfmt
+    - cargo fmt --all -- --check
+
+clippy:
+  stage:                           lint
+  <<:                              *docker-env
+  script:
+    - rustup component add clippy
+    - cargo clippy --all --verbose
+    - RUSTDOCFLAGS="--deny broken_intra_doc_links" cargo doc --verbose --workspace --no-deps --document-private-items
+
+#### stage:                       test
+
+check-code:
+  stage:                           test
+  <<:                              *docker-env
+  script:
+    - cargo check --all-targets
+    - cargo check --manifest-path http-client/Cargo.toml
+    - cargo check --manifest-path http-server/Cargo.toml
+    - cargo check --manifest-path ws-client/Cargo.toml
+    - cargo check --manifest-path ws-server/Cargo.toml
+    - cargo check --manifest-path types/Cargo.toml
+    - cargo check --manifest-path utils/Cargo.toml
+    - cargo check --manifest-path proc-macros/Cargo.toml
+    - cargo check --manifest-path test-utils/Cargo.toml
+    - cargo check --manifest-path examples/Cargo.toml
```

Comment:
@niklasad1 any proof to that? `--all` is deprecated to `--workspace`, is there any bug with running it?

Reply:
`--all` and `--workspace` are semantically the same AFAIK and doesn't fix anything but probably better to use `--workspace` 

The issue is that we have been faced is that the features in the workspace are leaked when compiling individual depending how exactly the used and with which features that are enabled crates such as `cargo check -p jsonrpsee-utils --features server`

However, I realize now that we should just do:

```bash
$ cargo hack check --workspace --each-feature
```

then it will work

Reply:
@alvicsam sorry I was wrong.

If you could just `cargo install cargo-hack &&  cargo hack check --workspace --each-feature` instead to would this